src/Microsoft.ML.PCA/PcaTrainer.cs

```diff
@@ -476,13 +476,13 @@ protected override void SaveCore(ModelSaveContext ctx)
             if (_mean.IsDense) // centered
             {
                 writer.WriteBoolByte(true);
-                writer.WriteFloatsNoCount(_mean.Values, _dimension);
+                writer.WriteSinglesNoCount(_mean.GetValues().Slice(0, _dimension));
```

Comment:
>_mean.GetValues().Slice(0, _dimension) [](start = 43, length = 38)

`_mean = VBufferUtils.CreateEmpty<float>(_dimension);`
`_mean = new VBuffer<float>(_dimension, meanArray);`

I think you don't have you use Slice, and can just pass whole span.

Reply:
`_mean` may have been passed in using the `internal PcaPredictor(IHostEnvironment env, int rank, VBuffer<float>[] eigenVectors, in VBuffer<float> mean)` constructor. It just seems safer to continue only using `_dimension` number of these elements. The existing code should have been using `_mean.Values, _mean.Count` otherwise.